src/k2/transport/RPCDispatcher.cpp

```diff
@@ -150,43 +157,62 @@ void RPCDispatcher::_handleNewMessage(Request&& request) {
     }
 }
 
-void RPCDispatcher::_send(Verb verb, std::unique_ptr<Payload> payload, TXEndpoint& endpoint, MessageMetadata meta) {
 
+seastar::future<>
+RPCDispatcher::_send(Verb verb, std::unique_ptr<Payload> payload, TXEndpoint& endpoint, MessageMetadata meta) {
+    auto ep = RPC().getServerEndpoint(endpoint.getProtocol());
```

Comment:
we are in the RPCDispatcher object here - no need to call RPC(). first. Moreover, this is the same endpoint as serverep below

Reply:
Used serverep = proto->getServerEndpoint() directly instead of calling RPC()